src/contracts/core/DelegationManager.sol

```diff
@@ -510,15 +534,23 @@ contract DelegationManager is
         delegatedTo[staker] = operator;
         emit StakerDelegated(staker, operator);
 
-        (IStrategy[] memory strategies, uint256[] memory shares) = getDelegatableShares(staker);
+        (IStrategy[] memory strategies, uint256[] memory shares) = getDelegatableScaledShares(staker);
+
+        // NOTE: technically the function returns scaled shares, so it needs to be descaled before scaling to the
+        // the new operator according to their totalMagnitude. However, since the staker is not delegated,
+        // scaledShares = shares so we can skip descaling and read these as descaled shares.
+        uint256[] memory scaledShares = ShareScalingLib.scaleShares(avsDirectory, operator, strategies, shares);
```

Comment:
we don't need to scale here

Reply:
We do, the staker is not currently delegated to anybody so their "scaledShares" aren't scaled accordingly to the new operator they are delegating to